sdks/java/core/src/main/java/org/apache/beam/sdk/util/common/Counter.java

```diff
@@ -295,6 +296,56 @@ public AggregationKind getKind() {
   public abstract CounterMean<T> getMean();
 
   /**
+   * CommitState represents whether counters' values have been committed to the backend.
+   *
+   * <p>Runners can use this information to optimize counters updates.
+   * For example, if counters are committed, runners may choose to skip the updates.
+   *
+   * <p>Counters' state transition table:
```

Comment:
Why does this need to be added to the SDK level model of Counters? It seems like this is a runner optimization for sending counter updates, so the implementation could be pushed to the runner/worker side.